src/Storefront/Resources/app/storefront/src/component-system/README.md

```diff
@@ -0,0 +1,361 @@
+
+In your component you can emit events to inform others about an action and pass additional data via the event.
+
+```javascript
+// MyComponent.js
+
+class MyComponent extends ShopwareComponent {
+
+    // ...
+
+    doSomething() {
+        const message = 'Hello World!';
+
+        window.Shopware.emit('MyComponent:DoSomething', message);
+    }
+}
+```
+
+Other components can the subscribe to this event to react on that.
+
+```javascript
+// SomeOtherComponent.js
+
+class SomeOtherComponent extends ShopwareComponent {
+
+    init() {
+        window.Shopware.on('MyComponent:DoSomething', (message) => {
+            this.el.innerText = message;
+        });
+    }
+}
+```
+
+Of course, you can also register to events from anywhere else, also from outside of the component system. For example, if you just want to extend the logic of an existing component.
+
+### Event Interception
+
+In addition to the normal asynchronous events, there is a seprate event type which expects a return value that gets further processed within the component. These events make it even easier to extend a components logic and offers a bunch of different use cases, like manipulating request data before it gets send. 
+
+For example the BuyButton component offers an event `BuyButton:PreSubmit` which is interceptable, as it is called via `emitInterception()`. It is triggered when a user clicks the buy button of a product.
+
+```javascript
+// BuyButton.js
+
+class BuyButton extends ShopwareComponent {
+
+    // ...
+
+    onFormSubmit(event) {
+        event.preventDefault();
+
+        let requestUrl = this.el.getAttribute('action');
+        let formData = window.Shopware.serializeForm(this.el);
+
+        [ requestUrl, formData ] = window.Shopware.emitInterception('BuyButton:PreSubmit', requestUrl, formData);
+
+        window.Shopware.emit('BuyButton:Submit', requestUrl, formData);
+
+        window.Shopware.callPluginMethod('OffCanvasCart', 'openOffCanvas', requestUrl, formData);
+    }
+}
+```
+
+You can see that the event `BuyButton:PreSubmit` offers the opportunity to manipulate the `formData` before it gets sent. From any other script you can intercept this event and work with the parameters from the event.
+
+```javascript
+// Intercept the buy button event
+window.Shopware.intercept('BuyButton:PreSubmit', (requestUrl, formData) => {
+    
+    formData.append('foo', 'bar');
+
+    return [ requestUrl, formData ];
+});
+```
+
+Don't forget to return the passed parameters in the same order, so the component logic can work with it. 
+
+There can be multiple subscribers to a single event. They will all be executed in the order as they are registered. You can change the order by passing a priority parameter as an optional third option, when registering an event. By default all subscribers have the priority `0`. The higher the priority the earlier the subscriber is called in the chain. Also negative values are possible to move a subscriber further down the chain.
```

Comment:
When all props are `0`, are they executed in the order in which they occur in the code? So synchronously top->bottom?